packages/date-time-picker/src/vaadin-date-time-picker.js

```diff
@@ -449,87 +469,98 @@ class DateTimePicker extends FieldMixin(SlotMixin(DisabledMixin(ThemableMixin(El
   }
 
   /** @private */
-  __datePickerChanged() {
-    const datePicker = this._getDirectSlotChild('date-picker');
-    if (this.__datePicker === datePicker || !datePicker) {
+  __onDomChange(addedNodes) {
```

Comment:
Let's add a TODO to restore the auto-generated pickers if the custom ones are removed.
I'd like to handle that separately with some kind of common helper.